calendar/index.js

```diff
@@ -0,0 +1,273 @@
+import { 
+    createMonthElement, 
+    createDayButton 
+} from './utils/dom.js'; 
+
+import { 
+    getNext12Months, 
+    formatDate
+} from './utils/dates.js'; 
+
+import { CLASS_NAMES } from './utils/constants.js';  
+
+import { 
+    dispatchSyncEvent, 
+    handleSyncEvent, 
+    saveToSessionStorage, 
+    getFromSessionStorage, 
+    clearSessionData, 
+    toggleVisibility, 
+    highlightButton, 
+    clearHighlights,
+    loadAndInjectCSS
+} from '../utils/components.js';  // Utility functions for general component behavior
+
+class IRNMNCalendar extends HTMLElement {
+    constructor() {
+        super();
+        this.label = this.getAttribute('label') || 'Check-in';
```

Comment:
@samuelsilvapt I'm thinking it might be worth splitting the init of properties to be handled as a seperate dedicate function that we call first in the connectedCallback function before running any of the rendering. That way we can keep the constructor cleaner. What do you think?